src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -96,7 +96,7 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
             timestampSeconds
         )
         logger.info { "h:$paymentHash generated payment request ${pr.write()}" }
-        db.addIncomingPayment(paymentPreimage, IncomingPayment.Origin.Invoice(pr))
+        db.addIncomingPayment(paymentPreimage, IncomingPayment.Origin.Invoice(pr), timestampSeconds)
```

Comment:
Nice catch! It don't see why we would not want to use the  `timestampSeconds` parameters here, @dpad85 do you agree ?

Reply:
The `createdAt` field must be in millis, not seconds. 

Also we could simply add a `delay(100)` in-between the `createInvoice` calls in the test, to make sure the ordering is always the same. I think it's a better solution than modifying the payment handler logic, where we would now have a special rule for incoming payments using LN invoices where the invoice's timestamp somehow matters. The `createdAt` timestamp is not very useful in the end, so it's not really worth creating this rule.

Reply:
Gotcha. I'll modify the test instead to ensure the ordering is correct. I see now why it makes sense that the expected behavior is for `addIncomingPayment` to use the default current time and not the `createAt` of the invoice.

Reply:
fixed in [9fff056](https://github.com/ACINQ/lightning-kmp/pull/338/commits/9fff0563b30e009f255e4560c9e07be37d18a865)

Reply:
Thanks!